src/main/java/studia/restControlers/RoomController.java

```diff
@@ -0,0 +1,95 @@
+package studia.restControlers;
+
+import com.google.cloud.firestore.*;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.annotation.*;
+import io.micronaut.security.annotation.Secured;
+import io.micronaut.security.rules.SecurityRule;
+import studia.datatypes.ReservationData;
+import studia.service.Firebase;
+
+import java.security.Principal;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import com.google.api.core.ApiFuture;
+
+@Secured(SecurityRule.IS_AUTHENTICATED)
+@Controller("/rooms")
+public class RoomController {
+
+    @Inject private Firebase firebase;
+
+    @Produces(MediaType.APPLICATION_JSON)
+    @Get("/my-rooms")
+    public List<Object> MyRooms() throws InterruptedException, ExecutionException {
+        Firestore db = firebase.getDb();
+
+        // Query query = db.collection("rooms").whereEqualTo("user", principal.getName());
+        Query query = db.collection("rooms")
+                .whereEqualTo("user", "mboruwa");
+
+        ApiFuture<QuerySnapshot> querySnapshot = query.get();
+        QuerySnapshot documents = querySnapshot.get();
```

Comment:
Nazwy tych zmiennych są mało intuicyjne imo, plus nie wiem po co to jest rozbite na dwie zmienne

Reply:
+